src/basis/SegmentTree.ml

```diff
@@ -32,9 +36,13 @@ struct
     | Some (r, v) ->
       if Pos.compare r.stop pos >= 0 then Some v else None
 
+  let containing pos t =
+    (* FIXME: This is suboptimal *)
+    CCList.of_iter @@ SegTree.values @@ SegTree.filter (fun range _ -> range.start <= pos && pos <= range.stop) t
```

Comment:
This might not be correct, because the "same" span can show up as multiple segments, and "shadowed" spans will not even be in the tree.

Reply:
I would suggest eventually removing `containing` completely---or we have to rethink the entire framework.